frontend/src/components/PaypalPreapproval.js

```diff
@@ -2,43 +2,46 @@ import React, { Component, PropTypes } from 'react';
 
 import ConnectPaypalButton from './ConnectPaypalButton';
 
-class PaypalReminder extends Component {
+class PaypalPreapproval extends Component {
   render() {
     const {
       card,
       onClickConnect,
       onClickInProgress,
-      i18n
+      i18n,
+      loading
     } = this.props;
 
     return (
-      <div className='PaypalReminder'>
-        {!card &&
+      <div className='PaypalPreapproval'>
+        {loading && <img src='/public/images/loading.gif' />}
+        {!card && !loading && 
           <div>
-            { i18n.getString('paypalReminderNeedValidAccount') }
+            { i18n.getString('paypalPreapprovalNeedValidAccount') }
             <ConnectPaypalButton
               disabled={ onClickInProgress }
               onClick={ onClickConnect }
               inProgress={ onClickInProgress }
               i18n={i18n}
             />
           </div>}
-        {card && 
+        {card && !loading &&
           <div>
-          { `${i18n.getString('paypalReminderAccountLinked')} ${card.number}.`} 
-          <span className='PaypalReminder-switch ml1' onClick={ onClickConnect }> {i18n.getString('paypalReminderRefreshKey')} </span>
+          { `${i18n.getString('paypalPreapprovalAccountLinked')} ${card.number}.`} 
+          <span className='PaypalPreapproval-switch ml1' onClick={ onClickConnect }> {i18n.getString('paypalPreapprovalRefreshKey')} </span>
           </div>}
 
       </div>
     );
   }
 }
 
-PaypalReminder.propTypes = {
-  card: PropTypes.object,
+PaypalPreapproval.propTypes = {
+  i18n: PropTypes.object.isRequired,
   onClickConnect: PropTypes.bool.isRequired,
+  card: PropTypes.object,
   onClickInProgress: PropTypes.bool,
-  i18n: PropTypes.object.isRequired,
+  loading: PropTypes.bool,
```

Comment:
The component doesn't know when we fetch the paypalCard, so how does it distinguish between no paypal card and loading state on its own?